### PR TITLE
Handle entity data packets.

### PR DESCRIPTION
**Pull request by julien-me**
- [x] Add a stub method to read entity data packets when an entity data packet is received.
- [x] Emit a signal after an entity data packet is read.

### CODE REVIEW

File: src/EntityServer.ts [hunk 1]
```diff
@@ -44,6 +45,7 @@ import AssignmentClient from "./domain/AssignmentClient";
  *
  *  @property {number} maxOctreePacketsPerSecond - The maximum number of octree packets per second that the user client is
  *      willing to handle.
+ *  @property {Signal} addedEntity - Triggered when an entity data packet is received.
```
Comment (ctrlaltdavid) [inline src/EntityServer.ts:48]:
```suggestion
 *  @property {Signal<EntityServer~AddedEntityCallback>} addedEntity - Triggered when an entity data packet is received.
```
See the `MessageMixer.messageReceived` JSDoc as an example.

File: src/EntityServer.ts [hunk 2]
```diff
@@ -117,6 +124,15 @@ class EntityServer extends AssignmentClient {
         return this.#_maxOctreePPS;
     }
 
+    /*@devdoc
+     *  Triggered when an entity data packet is received.
+     *  @function EntityServer.addedEntity
+     *  @returns {Signal}
```
Comment (ctrlaltdavid) [inline src/EntityServer.ts:130]: Document as `@callback` without a `@returns`. See the MessageMixer for example.

File: src/domain/networking/packets/PacketScribe.ts [hunk 3]
```diff
@@ -72,6 +73,8 @@ import BulkAvatarTraitsAck from "./BulkAvatarTraitsAck";
  *      {@link PacketScribe.AvatarIdentity&period;write|AvatarIdentity&period;write}
  *  @property {function} DomainConnectRequest.write -
  *      {@link PacketScribe.DomainConnectRequest&period;write|DomainConnectRequest&period;write}
+ *  @property {function} EntityData.read -
+ *      {@link PacketScribe.EntityData&period;read|EntityData&period;read}
```
Comment (ctrlaltdavid) [inline src/domain/networking/packets/PacketScribe.ts:77]: The link in the doc isn't working...
image

File: src/domain/networking/packets/EntityData.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+
+const EntityData = new class {
+    // C++ N/A
+
+    read(): void { /* eslint-disable-line class-methods-use-this */ }
```
Comment (ctrlaltdavid) [inline src/domain/networking/packets/EntityData.ts:15]: JSDoc needed.

File: src/domain/octree/OctreePacketProcessor.ts [hunk 5]
```diff
@@ -22,6 +24,7 @@ import ReceivedMessage from "../networking/ReceivedMessage";
  *
  *  @class OctreePacketProcessor
  *  @property {string} contextItemType="OctreePacketProcessor" - The type name for use with the {@link ContextManager}.
+ *  @property {Signal} Triggered when an entity data packet is received.
```
Comment (ctrlaltdavid) [inline src/domain/octree/OctreePacketProcessor.ts:27]:
```suggestion
 *  @property {Signal} addedEntity - Triggered when an entity data packet is received.
```

File: src/EntityServer.ts [hunk 6]
```diff
@@ -13,6 +13,7 @@ import NodeList from "./domain/networking/NodeList";
 import NodeType, { NodeTypeValue } from "./domain/networking/NodeType";
 import OctreeConstants from "./domain/octree/OctreeConstants";
 import OctreePacketProcessor from "./domain/octree/OctreePacketProcessor";
+import SignalEmitter, { Signal } from "./domain/shared/SignalEmitter";
 import OctreeQuery from "./domain/octree/OctreeQuery";
 import PacketScribe from "./domain/networking/packets/PacketScribe";
```
Comment (ctrlaltdavid) [inline src/EntityServer.ts:18]: Fix up order: networking, then octree, then shared.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: src/EntityServer.ts (hunk 1 above)
Comment (julien-me) [reply src/EntityServer.ts]: Fixed.

File: src/EntityServer.ts (hunk 2 above)
Comment (julien-me) [reply src/EntityServer.ts]: Fixed.

File: src/domain/networking/packets/PacketScribe.ts (hunk 3 above)
Comment (julien-me) [reply src/domain/networking/packets/PacketScribe.ts:78]: Fixed.

File: src/domain/networking/packets/EntityData.ts (hunk 4 above)
Comment (julien-me) [reply src/domain/networking/packets/EntityData.ts]: Fixed.

File: src/domain/octree/OctreePacketProcessor.ts (hunk 5 above)
Comment (julien-me) [reply src/domain/octree/OctreePacketProcessor.ts]: Fixed.

File: src/EntityServer.ts (hunk 6 above)
Comment (julien-me) [reply src/EntityServer.ts]: Fixed.

File: src/EntityServer.ts [hunk 7]
```diff
@@ -117,6 +125,14 @@ class EntityServer extends AssignmentClient {
         return this.#_maxOctreePPS;
     }
 
+    /*@devdoc
```
Comment (ctrlaltdavid) [inline src/EntityServer.ts:128]:
```suggestion
    /*@sdkdoc
```

File: src/domain/networking/packets/EntityData.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+
+type EntityDataDetails = {
+    // WEBRTC TODO: Address further code.
+};
+
+
+const EntityData = new class {
```
Comment (ctrlaltdavid) [inline src/domain/networking/packets/EntityData.ts:18]: The EntityData packet needs to be added to the packet list documentation in PacketHeaders.ts:
image

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: src/EntityServer.ts (hunk 7 above)
Comment (julien-me) [reply src/EntityServer.ts]: Fixed.

File: src/domain/networking/packets/EntityData.ts (hunk 8 above)
Comment (julien-me) [reply src/domain/networking/packets/EntityData.ts:18]: Added.

**Review by ctrlaltdavid — APPROVED** [review]

**Pull request merged**